src/System.Runtime.Serialization.Xml/tests/XmlDictionaryWriterTest.cs

```diff
@@ -164,6 +165,288 @@ public static void XmlDictionaryWriter_InvalidUnicodeChar()
+                {
+                    xreader.Read();
+                    xreader.ReadOuterXml();
+                }
+            }
+        }
+    }
+
+    [Fact]
+    public static void CreateTextReaderWriterTest()
+    {
+        string expected = "<localName>the value</localName>";
+        using (MemoryStream stream = new MemoryStream())
+        {
+            using (XmlDictionaryWriter writer = XmlDictionaryWriter.CreateTextWriter(stream, Encoding.UTF8, false))
+            {
+                writer.WriteElementString("localName", "the value");
+                writer.Flush();
+                byte[] bytes = stream.ToArray();
+                StreamReader reader = new StreamReader(stream);
+                stream.Position = 0;
+                string content = reader.ReadToEnd();
+                Assert.Equal(expected, content);
+                reader.Close();
+
+                using (XmlDictionaryReader xreader = XmlDictionaryReader.CreateTextReader(bytes, new XmlDictionaryReaderQuotas()))
+                {
+                    xreader.Read();
+                    string xml = xreader.ReadOuterXml();
+                    Assert.Equal(expected, xml);
+                }
+            }
+        }
+    }
+
+    [Fact]
+    public static void StreamProvoiderTest()
+    {
+        List<string> ReaderWriterType = new List<string>
+            {
+                "Binary",
+                //"MTOM", //MTOM methods not supported now.
+                //"MTOM",
+                //"MTOM",
+                "Text",
+                "Text",
+                "Text"
+            };
+
+        List<string> Encodings = new List<string>
+            {
+                "utf-8",
+                "utf-8",
+                "utf-16",
+                "unicodeFFFE",
+                "utf-8",
+                "utf-16",
+                "unicodeFFFE"
+            };
+
+        for (int i = 0; i < ReaderWriterType.Count; i++)
+        {
+            string rwTypeStr = ReaderWriterType[i];
+            ReaderWriterFactory.ReaderWriterType rwType = (ReaderWriterFactory.ReaderWriterType)
+            Enum.Parse(typeof(ReaderWriterFactory.ReaderWriterType), rwTypeStr, true);
+            Encoding encoding = Encoding.GetEncoding(Encodings[i]);
+
+            Random rndGen = new Random();
+            int byteArrayLength = rndGen.Next(100, 2000);
+            byte[] byteArray = new byte[byteArrayLength];
+            rndGen.NextBytes(byteArray);
+            MyStreamProvider myStreamProvider = new MyStreamProvider(new MemoryStream(byteArray));
+            bool success = false;
+            bool successBase64 = false;
+            MemoryStream ms = new MemoryStream();
+            success = WriteTest(ms, rwType, encoding, myStreamProvider);
+            Assert.True(success);
+            success = ReadTest(ms, encoding, rwType, byteArray);
+            Assert.True(success);
```

Comment:
Please add error message for calls to Assert.True(). It would be very hard to determine which Assert.True is failing without error message.